kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/TlsSecrets.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.operator;
+
+import java.nio.file.Path;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxy;
+
+public record TlsSecrets(Path path, String volumeName, String secretName) {
```

Comment:
Is there some validation we can do, that the certs have the expected SANs?

Reply:
have had a play around and yes, we could load the Secret and parse `tls.crt`. A little string manipulation is required since JDK doesn't understand PEM format, but you can load it into a `java.security.cert.X509Certificate` and get at the SANs.